### DISCUSSION TITLE
High CPU Usage / .net 6 / linux

### DISCUSSION THREAD

**Discussion by Stan92** [rabbitmq/rabbitmq-dotnet-client#1241] [Q&A] [answered]
Hi,

I'm running around 30 .net core 6 workers and I notice there's a huge CPU usage (my server is close to 100% cpu usage).
I stopped 10 of them, but the cpu usage remains the same except that the active workers use more CPU.

The context.
RabbitMQ Cluster (3 nodes) on 3 remote servers.
All the workers run on a  root vps (10 vCore, 64g Ram) (Ubuntu 22.04)
RabbitMQ.Client 6.4.0
I use quorum queue

Did I miss something?

This is how look like my consumers (I think they are very simple).

```
private async Task CoreProcessAsync(int terminaison, CancellationToken stoppingToken)
        {
            await Task.Delay(1, stoppingToken);
            int nombreMaxi = _parametres.NbItemsACharger;
            string queueName = $"{RABBITMQ_QUEUES_ROUTING.MQTT}-{terminaison}";
            var consumerAsync = new AsyncEventingBasicConsumer(_rabbitChannel);
            consumerAsync.Received += async (model, ea) =>
            {
                var body = ea.Body.ToArray();
                var message = Encoding.UTF8.GetString(body);
                var item = JsonSerializer.Deserialize<MqttFluxBoitier>(message);
                MqttFlux myMqttItem = new MqttFlux
                {
                    Topic = item.Topic,
                    Payload = item.Payload
                };
                await SendAsync(myMqttItem);
                _rabbitChannel.BasicAck(ea.DeliveryTag, false);
            };
            _rabbitChannel.BasicQos(0, (ushort)nombreMaxi, false);
            basicConsumerTag = _rabbitChannel.BasicConsume(queue: queueName, autoAck: false, consumer: consumerAsync);
            while (!stoppingToken.IsCancellationRequested);
            _logger.ForContext("Terminaison", terminaison).Warning("Déconnexion");
}


```
top -i result with all the workers
![b1](https://user-images.githubusercontent.com/1044972/187089467-27a437ef-f3b0-4337-b0d4-2d6cb23f590d.PNG)

top -i result when I stop 10 workers
![b2](https://user-images.githubusercontent.com/1044972/187089498-51700760-fa6e-4278-83b7-a6db6dcb4cbb.PNG)

the same application using Redis Streams
![b3](https://user-images.githubusercontent.com/1044972/187089524-bccfd9e3-2421-4db2-a3a9-216cdac69498.PNG)

**Comment by bording** [comment, marked as answer]
> `while (!stoppingToken.IsCancellationRequested);`

Checking the value of `IsCancellationRequested` is not a blocking call, so that loop is spinning endlessly and consuming all your cpu resources.

Why are you trying to keep that method alive? What you are doing there is not going to work.
